### PR TITLE
implement helm get values with sysdump

### PR DESCRIPTION
**Pull request by liyihuang**

### CODE REVIEW

File: sysdump/defaults.go [hunk 1]
```diff
@@ -15,6 +16,7 @@ const (
 const (
 	DefaultCiliumLabelSelector               = labelPrefix + "cilium"
 	DefaultCiliumEnvoyLabelSelector          = labelPrefix + "cilium-envoy"
+	DefaultCiliumHelmChartName               = defaults.HelmReleaseName
```
Comment (liyihuang) [inline sysdump/defaults.go]: https://github.com/liyihuang/cilium-cli/commit/f093eec1c5fee350d8b900d62341b926da4932b8#r123316956


follow the discussion from here


If that's the case, is it better just to follow what has been done like other parameters?

File: k8s/client.go [hunk 2]
```diff
@@ -1084,6 +1085,23 @@ func (c *Client) GetHelmState(ctx context.Context, namespace string, secretName
...
+// GetHelmValues is the function for cilium cli sysdump to collect the helm values from the release directly
+func (c *Client) GetHelmValues(_ context.Context, releaseName string) ([]byte, error) {
+	client := action.NewGetValues(c.HelmActionConfig)
+	vals, err := client.Run(releaseName)
+	if err != nil {
+		return nil, fmt.Errorf("unable to retrive the helm value from the release %s: %w", releaseName, err)
+	}
+
+	valuesBuf := new(bytes.Buffer)
+	encodeErr := output.EncodeYAML(valuesBuf, vals)
```
Comment (liyihuang) [inline k8s/client.go]: follow the conversation from here

https://github.com/liyihuang/cilium-cli/commit/f093eec1c5fee350d8b900d62341b926da4932b8#r123316401


If that's the case, why the following can be reused with a new declare?


https://github.com/liyihuang/cilium-cli/blob/f093eec1c5fee350d8b900d62341b926da4932b8/k8s/client.go#L1067-L1077

File: k8s/client.go [hunk 3]
```diff
@@ -1084,6 +1085,23 @@ func (c *Client) GetHelmState(ctx context.Context, namespace string, secretName
 	}, nil
 }
 
+// GetHelmValues is the function for cilium cli sysdump to collect the helm values from the release directly
+func (c *Client) GetHelmValues(_ context.Context, releaseName string) ([]byte, error) {
```
Comment (liyihuang) [inline k8s/client.go]: follow the conversation https://github.com/liyihuang/cilium-cli/commit/f093eec1c5fee350d8b900d62341b926da4932b8#r123316881

I do see other function dont pass it. like this one

https://github.com/liyihuang/cilium-cli/blob/f093eec1c5fee350d8b900d62341b926da4932b8/k8s/client.go#L1105

File: sysdump/defaults.go (hunk 1 above)
Comment (asauber) [reply sysdump/defaults.go]: no, it's best to use the defaults package here, that's what we expect elsewhere in the CLI for the release name

at some point we can also take the release name from a flag

File: k8s/client.go (hunk 2 above)
Comment (asauber) [reply k8s/client.go]: If there are _any_ new names on the left side of a `:=`, then the `:=` is allowed.

**Comment by liyihuang** [comment]
the example of the output file.

the current output is just user provided value.  

In the future, we can provide more advanced options(like get some other version of the release or get the whole value file with out default values)
[helm-values-20230801-150055.txt](https://github.com/liyihuang/cilium-cli/files/12233358/helm-values-20230801-150055.txt)

File: k8s/client.go (hunk 3 above)
Comment (asauber) [reply k8s/client.go]: Ah, I see. Then this is fine. This appears to be in anticipation of some future method calls which might use the context, which I personally don't agree with including here, but given the style of the file, it's fine to keep like this.

File: internal/cli/cmd/sysdump.go [hunk 4]
```diff
@@ -74,6 +74,9 @@ func initSysdumpFlags(cmd *cobra.Command, options *sysdump.Options, optionPrefix
 	cmd.Flags().StringVar(&options.CiliumEnvoyLabelSelector,
 		optionPrefix+"cilium-envoy-label-selector", sysdump.DefaultCiliumEnvoyLabelSelector,
 		"The labels used to target Cilium Envoy pods")
+	cmd.Flags().StringVar(&options.CiliumHelmChartName,
```
Comment (asauber) [inline internal/cli/cmd/sysdump.go]: This `CiliumHelmChartName` isn't used as the chart name, but rather the release name. So this flag and the `sysdump.DefaultCiliumHelmChart` name should instead be `..ReleaseName..`

**Pull request closed without merging**